packages/flutter_tools/lib/src/commands/create_base.dart

```diff
@@ -784,12 +784,31 @@ bool isValidPackageName(String name) {
       !_keywords.contains(name);
 }
 
+/// Returns a potential valid name from the given [name].
+@visibleForTesting
+String potentialValidPackageName(String name){
+  String newName = name.toLowerCase();
+  if (newName.startsWith(RegExp(r'[0-9]'))) {
+    // If the package starts with a number, prepend '_'.
+    newName = '_$newName';
+  }
+  // Replaces all the non-alphanumeric characters with '_' and join the
+  // consecutive '_'s.
+  return newName.replaceAll(RegExp(r'[^a-z0-9]+'), '_');
+}
+
 // Return null if the project name is legal. Return a validation message if
 // we should disallow the project name.
 String? _validateProjectName(String projectName) {
   if (!isValidPackageName(projectName)) {
-    return '"$projectName" is not a valid Dart package name.\n\n'
-        'See https://dart.dev/tools/pub/pubspec#name for more information.';
+    final String potentialValidName = potentialValidPackageName(projectName);
+
+    return <String>[
+      '"$projectName" is not a valid Dart package name.',
+      if (isValidPackageName(potentialValidName)) '\nTry "$potentialValidName" instead.',
```

Comment:
Nitpick:

I am not sure I like that `potentialValidPackageName` can return a name that's still invalid. That could be confusing given its name.

Rather than checking `potentialValidName` with `isValidPackageName`, how about having `potentialValidPackageName` do the check instead? Then, if the new name that `potentialValidPackageName` finds still isn't valid, it can instead return `null`.

Reply:
Sure that makes sense, I made the change in [refactor: Returns null if the name is not valid and only replace hyphens](https://github.com/flutter/flutter/pull/130900/commits/a5de39de661fdb0de9192b15fdfcbe15e02ec504)